DataDog/datadog-agent
# Do not store api key in retry file

### What does this PR do?

This PR avoids storing the API key in `.retry` files.

### Additional Notes

If `agentchecks` transactions can use `TransactionPriorityHigh`, then `SubmitAgentchecksMetadata` and `SendAgentchecksMetadata` can be removed.

### Describe your test plan
- Set `forwarder_retry_queue_payloads_max_size` to 50
- Set `forwarder_storage_max_size_in_bytes` to `50000000`
- Run the Agent so that there are more than 1 hour of `retry` files (to make sure all metadata requests are sent)
- Check the API key is not stored in any `.retry` files (Also check API Key is not stored in the compressed payload). I can provide a tool to dump the `retry_file` to json.
- Check the domain is not stored in any `.retry` files.
- Check `HostMetadata ` and `AgentChecks` are not stored on disk. 
- Check a `.retry` file can be reloaded on Windows.
- Check the behavior when decreasing the number of API keys for the same domain.
- Check `.retry` file cannot be edited by a regular user (Windows + Unix).
